lib/include/fdotypes.h

```diff
@@ -237,37 +237,63 @@ bool fdo_public_key_write(fdow_t *fdow, fdo_public_key_t *pk);
 fdo_public_key_t *fdo_public_key_read(fdor_t *fdor);
 fdo_public_key_t *fdo_public_key_clone(fdo_public_key_t *pk);
 
-#define AES_IV 16
-#define AES_CTR_IV 12
-#define AES_CTR_IV_COUNTER 4
+// AES GCM IV length
+#define AES_GCM_IV 12
+// AES CCM IV length
+#define AES_CCM_IV 7
+
+// AES GCM authenticated TAG length
+#define AES_GCM_TAG_LEN 16
+// AES CCM authenticated TAG length
+#define AES_CCM_TAG_LEN 16
+
+#if defined(AES_MODE_GCM_ENABLED)
+#define AES_IV_LEN AES_GCM_IV
+#define AES_TAG_LEN AES_GCM_TAG_LEN
+#elif defined(AES_MODE_CCM_ENABLED)
+#define AES_IV_LEN AES_CCM_IV
+#define AES_TAG_LEN AES_CCM_TAG_LEN
+#else
+// Update IV length for AES here, if AES CTR/CBC modes are needed as per the FDO specification
+#define AES_IV_LEN 16
```

Comment:
We can remove this here. We have removed the AES CTR/CBC support from many places, it might not be trivial add that support back.